harness/determined/cli/user.py

```diff
@@ -244,51 +243,51 @@ def edit(args: argparse.Namespace) -> None:
 
 
 def render_token_info(token_info: Sequence[bindings.v1TokenInfo]) -> None:
-    values = []
-    for ti in token_info:
-        value = [
-            ti.id,
-            ti.userId,
-            ti.description,
-            ti.createdAt,
-            ti.expiry,
-            ti.revoked,
-            ti.tokenType,
-        ]
-        values.append(value)
+    values = [
+        [t.id, t.userId, t.description, t.createdAt, t.expiry, t.revoked, t.tokenType]
+        for t in token_info
+    ]
     render.tabulate_or_csv(TOKEN_HEADERS, values, False)
 
 
-def print_token_info(data: Sequence[bindings.v1TokenInfo], args: argparse.Namespace) -> None:
-    if data:
-        json_data = [elem.to_json() for elem in data]
-        if len(data) == 1:
-            json_data = json_data[0]
-    else:
-        json_data = {}
-    if args.yaml:
-        print(util.yaml_safe_dump(json_data, default_flow_style=False))
-    else:
-        render.print_json(json_data)
-
-
 def describe_token(args: argparse.Namespace) -> None:
     sess = cli.setup_session(args)
     result = []
-    for arg in args.username:
-        userID = bindings.get_GetUserByUsername(session=sess, username=arg).user.id
-        resp = bindings.get_GetAccessToken(session=sess, userId=userID)
-        result.append(resp.tokenInfo)
-    render_token_info(result)
+    for token_id in args.token_id:
+        filter_data = json.dumps({"TokenId": token_id, "TokenType": "ACCESS_TOKEN"})
+        try:
+            resp = bindings.get_GetAccessTokens(session=sess, filter=filter_data)
+            if len(resp.tokenInfo) == 1:
+                result.append(resp.tokenInfo[0])
+            # We might want to catch exception here
```

Comment:
We might want to catch exception here

Reply:
Which exception are you thinking here?

Reply:
Is it not being caught by the except on 252? Or are we expecting something to go wrong with the Get request, like an un-authed/insufficient permissions?

Reply:
So I was convinced with the messages I have the `else` in line 250, and was looking for suggestion on how can we encapsulate all types of error in that message.

Reply:
Just to double check on this, the check here is:
* We loop over each provided token & we attempt to retrieve the AccessTokens based on the provided filter data (typically going to be 1+ token ids)
* If the length of the response is 1 then we append the results; otherwise we report that we didn't find the token?

Would it potentially make more sense to instead create a larger filter with all the results and check and make sure that the result set length matches the amount of tokens provided instead of making multiple API calls to retrieve information for each individual token?

Reply:
But also back to the original question in terms of how we want to handle responding to errors here; what responses are we wanting to make sure we give to users?
* Happy Path: Return described token records
* Less Than Happy Path #1: Unable to find some subset/all of the requested tokens because they don't exist
* LTHP #2: Unable to described requested tokens due to permissions (should likely give same response as LTHP#1)
* LTHP #3: Invalid argument (negative id value, etc), maybe something we can setup in the command to restrict the requested id's to positive integers; otherwise, we can handle this like LTHP #1 or returns the API error 
* LTHP #4: Internal server error/unexpected error, this is a more broad case and we might be able to leave these uncaught and have it reported to the user in that way (but it's worth double checking to make sure that's an acceptable path)

Reply:
Let me walk you through the `describe_token`

input: multiple token_id(s)
output: rendered token info(in json format)

1. loop through the given ids
2. create a filter data with the token id. 
3. As token ids are unique in the table, only 1 row is expected to be returned.
4. Errors that can arrive: 
- one of the token ids do not exist
- no permission to view one of the token ids
5. will add to a result list to get the output rendered together and return the output json

Let me know if it makes sense

Reply:
My concern with making individual requests for each token is that it might be inefficient if we're looking at querying for a lot of token ids since we'll be making multiple API calls for each individual token and handling each on individually; instead of we make a larger batched requested we might be able to improve performance and reduce the IO overhead.

Reply:
I understand your concern and completely agree.

In our case, the `filter_data` can't take multiple token IDs, and each token needs a separate request, one possible improvement is to parallelize the API calls or handle them more efficiently.

We could parallelize the API calls to speed things up using Python's concurrent.futures module

```
def fetch_token_info(sess, token_id):
    filter_data = json.dumps({
        "TokenId": token_id,
        "TokenType": "ACCESS_TOKEN"
    })
    try:
        resp = bindings.get_GetAccessTokens(session=sess, filter=filter_data)
        if len(resp.tokenInfo) == 1:
            return resp.tokenInfo[0]
        else:
            print(f"Could not find token with ID: {token_id}")
    except api.errors.NotFoundException:
        print(f"Token {token_id} not found")
    except Exception as e:
        print(f"Failed to fetch token {token_id}: {e}")
    return None

def describe_token(args: argparse.Namespace) -> None:
    sess = cli.setup_session(args)
    result = []
    
    # Use ThreadPoolExecutor to parallelize API calls
    with concurrent.futures.ThreadPoolExecutor() as executor:
        futures = [executor.submit(fetch_token_info, sess, token_id) for token_id in args.token_id]
        
        for future in concurrent.futures.as_completed(futures):
            token_info = future.result()
            if token_info:
                result.append(token_info)
    
    if result:
        render_token_info(result)
```

Reply:
Taking a step back, Is there anything preventing us from modifying the endpoint to accept multiple token id's in the filter data?

Reply:
Yea, we'd likely need to drop the username filter for this specific CLI request & then add in the appropriate logic at the API level to confirm that the requesting user has the appropriate permission to view all the requested tokens & filter out any that shouldn't be displayed

Reply:
That would be more inline with how we handle experiments/runs